### PR TITLE
fix: member listing pagination issue

### PR DESCRIPTION
**Pull request by iammuho**
Related Issue: https://github.com/mergeability/mergeable/issues/442

-- 

This PR follows the second idea which I've changed the alghorithm to fetch approved users membership states and if any of them are true, it will be counted.

### CODE REVIEW

File: lib/validators/options_processor/teams.js [hunk 1]
```diff
@@ -19,6 +19,26 @@ class Teams {
...
+        }
+
+        if (membershipState) { teamMembers.push(user) }
+      }
+    }
+    return _.uniq(teamMembers)
+  }
 }
 
 const getTeamMembers = async (context, team) => {
```
Comment (shine2lay) [inline lib/validators/options_processor/teams.js:44]: i see that `[owner](https://github.com/mergeability/mergeable/blob/master/lib/validators/options_processor/owners.js#L37)` file still uses this function, is there a particular reason for it?

**Review by shine2lay — CHANGES_REQUESTED** [review]

File: lib/validators/options_processor/teams.js [hunk 2]
```diff
@@ -47,4 +67,35 @@ const getTeamMembers = async (context, team) => {
...
+      username: username
+    })
+  } catch (err) {
+    if (err.status === 404) {
+      return false
+    }
+    throw err
+  }
+
+  if (res.data.state === 'active') {
```
Comment (jusx) [inline lib/validators/options_processor/teams.js:94]: You can replace L94 - 98 in a succinct manner:

` return (res.data.state === 'active')`

File: lib/validators/options_processor/teams.js [hunk 3]
```diff
@@ -47,4 +67,35 @@ const getTeamMembers = async (context, team) => {
   return res.data.map(member => member.login)
 }
 
+const getTeamMembershipState = async (context, team, username) => {
+  const stringArray = team.split('/')
+  if (stringArray.length !== 2) {
+    throw Error(`each team id needs to be given in 'org/team_slug'`)
+  }
+  if (stringArray[0].indexOf('@') === 0) stringArray[0] = stringArray[0].substring(1)
```
Comment (jusx) [inline lib/validators/options_processor/teams.js:75]: Looks like you are supporting both naming prefix with `@` and without here. i.e.`@org/teamName` and `org/teamName` here. 

Good call.

To remove the first element, you may want to consider using `stringArray.shift()` here.

**Comment by jusx** [comment]
Love the performant solution to this. Thank you.

File: lib/validators/options_processor/teams.js (hunk 3 above)
Comment (iammuho) [reply lib/validators/options_processor/teams.js:75]: In fact, I haven't add this but just copy/paste from old code. Do you want me to replace again ?

Comment (jusx) [reply lib/validators/options_processor/teams.js:75]: I do think it's more readable with `shift`. If you agree, please do. Not a big deal otherwise.

**Comment by iammuho** [comment]
hey @jusx ,

I've used the shift, can you re-review?

best

**Review by jusx — APPROVED** [review]

**Comment by jusx** [comment]
@iammuho I went ahead and merged this. It would be great if you are able to do a separate PR to add code coverage on the new function in `Teams` Thanks.

**Pull request merged**